### PR TITLE
[DataFactory] Add Encryption settings to Data Factory

### PR DESCRIPTION
**Pull request by Pacodel**
## Description

Added Encryption (CMK) to DataFactory

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/956
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by BethanyZhou** [comment]
Hi @Pacodel , have you proposed a design review on https://github.com/Azure/azure-powershell-cmdlet-review-pr? please link it with this pr.

File: src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs [hunk 1]
```diff
@@ -484,11 +505,23 @@ public override void ExecuteCmdlet()
             }
             FactoryIdentity factoryIdentity = new FactoryIdentity(factoryIdentityType, userAssignedIdentities: this.UserAssignedIdentity);
 
+            EncryptionConfiguration encryption = null;
+            if(!string.IsNullOrWhiteSpace(this.EncryptionVaultBaseUrl) && !string.IsNullOrWhiteSpace(this.EncryptionKeyName))
```
Comment (BethanyZhou) [inline src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs:509]: As the description in design doc, if VaultBaseUrl is present, then keyName needs to be present as well. Any check for this?

File: src/DataFactory/DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs [hunk 2]
```diff
@@ -35,6 +37,36 @@ public class UpdateAzureDataFactoryCommand : DataFactoryBaseCmdlet
         [ValidateNotNullOrEmpty]
         public string Name { get; set; }
 
+        #region Attributes
+        [Parameter(Mandatory = false, HelpMessage = Constants.HelpIdentityType)]
+        #endregion
+        public string IdentityType { get; set; }
+
+        #region Attributes
```
Comment (BethanyZhou) [inline src/DataFactory/DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs:45]: The change for Update-AzDataFatory doesn't mentioned in design doc. Please add it.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs (hunk 1 above)
Comment (Pacodel) [reply src/DataFactory/DataFactoryV2/DataFactories/SetAzureDataFactoryCommand.cs:509]: We are validating in backend and they'll get an error message. This is what we did for SDK as well to have a consistent experience.

File: src/DataFactory/DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs (hunk 2 above)
Comment (Pacodel) [reply src/DataFactory/DataFactoryV2/DataFactories/UpdateAzureDataFactoryCommand.cs:45]: Updated the issue: 
https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/956

**Comment by Pacodel** [comment]
@BethanyZhou here's the issue: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/956
I couldn't edit the description (it fails).

**Review by BethanyZhou — DISMISSED** [review]

File: src/DataFactory/DataFactoryV2/help/Update-AzDataFactoryV2.md [hunk 3]
```diff
@@ -15,18 +15,27 @@ Updates the properties of a data factory.
 ### ByFactoryName (Default)
 ```
 Update-AzDataFactoryV2 [-ResourceGroupName] <String> [-Name] <String> [-Tag <Hashtable>]
+ [-IdentityType] <String>
```
Comment (BethanyZhou) [inline src/DataFactory/DataFactoryV2/help/Update-AzDataFactoryV2.md:18]: IdentityType is optional why its syntax says it is required? Have you updated this markdown by https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/help-generation.md#updating-a-single-markdown-file?

Comment (Pacodel) [reply src/DataFactory/DataFactoryV2/help/Update-AzDataFactoryV2.md]: @BethanyZhou my team was always modifying that file manually. I have followed the steps for both markdown files and updated the PR

**Review by BethanyZhou — APPROVED** [review]

**Comment by kk921dbg** [comment]
Have tried the code for this change to add our CMK to a Data Factory that already exists, it doesnt work so not sure if it is implemented even though documentation says that it is. Can you let us know if it should actually work for adding CMK to existing ADF using Powershell? az cli does not have this feature either but does not say it does in its help doc unlike 
the powershell

**Pull request merged**